### PR TITLE
refac: Add metering point id to pos neg gl

### PR DESCRIPTION
**Pull request by johevemi**

### CODE REVIEW

File: source/databricks/calculation_engine/tests/datamigration/test_energy_results_migrations.py [hunk 1]
```diff
@@ -234,3 +241,71 @@ def test__result_table__is_not_managed(
...
+
+@pytest.mark.parametrize(
+    "metering_point_id,time_series_type",
+    [
+        ("571313180480500149", "negative_grid_loss"),
+        ("571313180480500149", "positive_grid_loss"),
+        (None, "production"),
+    ],
+)
+def test__migrated_table_special_metering_point_id_constraint_valid_data(
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/datamigration/test_energy_results_migrations.py:254]: The test name is not clear to me. What do you mean by 'special'?

File: source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py [hunk 2]
```diff
@@ -90,29 +91,66 @@ def calculate_grid_loss(
...
+        )
+        .distinct()
+        .where(
+            grid_loss_responsible.df[Colname.metering_point_type]
+            == metering_point_type.value
+        )
+    )
+
+
+def calculate_negative_grid_loss(
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py:109]: Is this new stuff tested?

File: source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py [hunk 3]
```diff
@@ -90,29 +91,66 @@ def calculate_grid_loss(
...
+    grid_loss: EnergyResults, grid_loss_responsible: GridLossResponsible
+) -> EnergyResults:
+    only_grid_area_and_metering_point_id = (
```
Comment (johevemi) [inline source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py:112]: rename to something like virtual_metering_point_per_grid_area_for_production

File: source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py [hunk 4]
```diff
@@ -90,29 +91,66 @@ def calculate_grid_loss(
...
+        _get_metering_point_id_and_grid_area_with_specific_metering_point_type(
+            grid_loss_responsible, MeteringPointType.PRODUCTION
+        )
+    )
+
+    result = grid_loss.df.join(
+        only_grid_area_and_metering_point_id, Colname.grid_area, "left"
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py:119]: I guess that here we assume that the grid area appears only once in only_grid_area_and_metering_point_id? If yes, then a comment would be appropriate (also in the positive grid loss method)

File: source/databricks/calculation_engine/tests/datamigration/test_energy_results_migrations.py (hunk 1 above)
Comment (johevemi) [reply source/databricks/calculation_engine/tests/datamigration/test_energy_results_migrations.py]: rename to  test__migrated_table_constraints_on_metering_point_id_with_valid_data

File: source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py (hunk 2 above)
Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py:109]: not directly

File: source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py (hunk 4 above)
Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/energy/aggregators/grid_loss_aggregators.py:115]: i need more details

File: source/databricks/calculation_engine/tests/datamigration/test_energy_results_migrations.py [hunk 5]
```diff
@@ -234,3 +241,71 @@ def test__result_table__is_not_managed(
...
 
     assert not table_location.startswith(database_location)
+
+
+@pytest.mark.parametrize(
+    "metering_point_id,time_series_type",
+    [
+        ("571313180480500149", "negative_grid_loss"),
+        ("571313180480500149", "positive_grid_loss"),
+        (None, "production"),
```
Comment (johevemi) [inline source/databricks/calculation_engine/tests/datamigration/test_energy_results_migrations.py:251]: add all time series types

**Review by energinet-ajw — APPROVED** [review]

**Pull request merged**